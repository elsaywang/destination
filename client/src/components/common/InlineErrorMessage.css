:local(.error) {
    color: var(--spectrum-semantic-negative-color-default);
    font-style: italic;
    display: flex;
    align-items: center;
}

:local(.errorMessage) {
    margin-left: 0.5rem;
}
