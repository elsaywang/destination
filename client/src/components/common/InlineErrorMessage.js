import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Alert from '@react/react-spectrum/Icon/Alert';
import styles from './InlineErrorMessage.css';

const InlineErrorMessage = ({ isInvalid, errorMessage, className }) => {
    if (isInvalid) {
        return (
            <span data-test="inline-error" className={classNames(styles.error, className)}>
                <Alert size="S" /> <span className={styles.errorMessage}>{errorMessage}</span>
            </span>
        );
    }

    return null;
};

InlineErrorMessage.propTypes = {
    isInvalid: PropTypes.bool.isRequired,
    errorMessage: PropTypes.string.isRequired,
    className: PropTypes.string,
};

export default InlineErrorMessage;
