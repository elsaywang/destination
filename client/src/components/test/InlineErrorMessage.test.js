import React from 'react';
import { shallow } from 'enzyme';
import InlineErrorMessage from '../../components/common/InlineErrorMessage';

describe('<InlineErrorMessage /> component', () => {
    const errorMessage = 'This is an error message.';

    describe('when isInvalid is true,', () => {
        const wrapper = shallow(
            <InlineErrorMessage isInvalid={true} errorMessage={errorMessage} />,
        );

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <InlineErrorMessage /> component with an error message', () => {
            expect(wrapper.find('[data-test="inline-error-message"]').exists()).toBe(true);
            expect(wrapper.text()).toBe(errorMessage);
        });
    });

    describe('when isInvalid is false', () => {
        const wrapper = shallow(
            <InlineErrorMessage isInvalid={false} errorMessage={errorMessage} />,
        );

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <InlineErrorMessage /> component with no error message', () => {
            expect(wrapper.find('[data-test="inline-error-message"]').exists()).toBe(false);
            expect(wrapper.text()).toBe('');
        });
    });
});
