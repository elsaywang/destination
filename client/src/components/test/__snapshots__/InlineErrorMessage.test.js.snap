// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineErrorMessage /> component when isInvalid is false matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InlineErrorMessage
    errorMessage="This is an error message."
    isInvalid={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<InlineErrorMessage /> component when isInvalid is true, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InlineErrorMessage
    errorMessage="This is an error message."
    isInvalid={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Alert
          size="S"
/>,
        <span
          className={undefined}
>
          This is an error message.
</span>,
      ],
      "className": "",
      "data-test": "inline-error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "S",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is an error message.",
          "className": undefined,
        },
        "ref": null,
        "rendered": "This is an error message.",
        "type": "span",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Alert
            size="S"
/>,
          <span
            className={undefined}
>
            This is an error message.
</span>,
        ],
        "className": "",
        "data-test": "inline-error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "S",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is an error message.",
            "className": undefined,
          },
          "ref": null,
          "rendered": "This is an error message.",
          "type": "span",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
